analyzers/tests/SonarAnalyzer.Test/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -172,6 +173,56 @@ public void ArrowExpressionBody_WithNotExpectedNode_ReturnsNull()
         public void GetDeclarationTypeName_Struct() =>
             ExtensionsCS.GetDeclarationTypeName(SyntaxFactory.StructDeclaration("MyStruct")).Should().Be("struct");
 
+        [TestMethod]
+        public void GetBody_FieldDeclaration()
+        {
+            ExtensionsCS.GetBody(SyntaxFactory.FieldDeclaration(SyntaxFactory.VariableDeclaration(SyntaxFactory.ParseTypeName("int")))).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_PropertyDeclaration_SingleAccessor()
+        {
+            var code = "class AClass { int AProperty { get => 42; } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<PropertyDeclarationSyntax>()).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_PropertyDeclaration_MultipleAccessors()
+        {
+            var code = "class AClass { int AProperty { get => 42; set { } } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<PropertyDeclarationSyntax>()).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_AccessorDeclaration_ArrowExpression()
+        {
+            var code = "class AClass { int AProperty { get => 42; } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<AccessorDeclarationSyntax>()).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_AccessorDeclaration_BodyBlock()
+        {
+            var code = "class AClass { int AProperty { set { var x1; var x2; var x3; } } }";
```

Comment:
Code should compile without errors.
```suggestion
            var code = "class AClass { int AProperty { set { var x1 = 0; var x2 = 0; var x3 = 0; } } }";
```

Reply:
As [above](https://github.com/SonarSource/sonar-dotnet/pull/8639#discussion_r1471171636).